File: api-express/src/controllers/cliente-controller.js
import { createClienteSchema, loginSchema } from "../validations/cliente-validations.js";
import * as clienteService from '../services/cliente-service.js'
import AppMessage from '../utils/app-message.js'

export default class ClienteController {
    static async create(req, res) {
        const payload = createClienteSchema.parse(req.body);

        const result = await clienteService.create(payload);
        return res.status(201).send(new AppMessage('Cliente criado com sucesso', 201, result));
    }

    static async login(req, res) {
        const payload = loginSchema.parse(req.body);

        const result = await clienteService.login(payload);
        return res.status(200).send(new AppMessage('Login realizado com sucesso', 200, result))
    }
}

File: api-actix/src/controllers/cliente_controller.rs
use actix_web::{post, web, HttpResponse, Result};
use crate::services::cliente_service::{ClienteService, LoginPayload};
use crate::utils::app_message::{success_response, ApiError};
use crate::db::AppState;
use crate::models::cliente::CreateClientePayload;

#[post("")]
async fn create(
    app_state: web::Data<AppState>,
    payload: web::Json<CreateClientePayload>
) -> Result<HttpResponse, ApiError> {
    let result = ClienteService::create(&app_state.db_pool, payload.into_inner()).await?;
    Ok(success_response("Cliente criado com sucesso", 201, result))
}

#[post("/login")]
async fn login(
    app_state: web::Data<AppState>,
    payload: web::Json<LoginPayload>
) -> Result<HttpResponse, ApiError> {
    let result = ClienteService::login(&app_state.db_pool, payload.into_inner()).await?;
    Ok(success_response("Login realizado com sucesso", 200, result))
}